var/spack/repos/builtin/packages/cp2k/package.py

```diff
@@ -97,6 +97,7 @@ class Cp2k(MakefilePackage, CudaPackage):
     depends_on('libxc@2.2.2:', when='+libxc@:5.5999', type='build')
     depends_on('libxc@4.0.3:', when='+libxc@6.0:6.9', type='build')
     depends_on('libxc@4.0.3:', when='+libxc@7.0:')
+    depends_on('libxc@4.3.4', type='link')
```

Comment:
Wouldn't something like this make more sense (or whatever range is appropriate for the older versions)?
```python
depends_on('libxc@2.2.2:4.99.0', when='+libxc@:5.5999', type='build')
depends_on('libxc@4.0.3:4.99.0', when='+libxc@6.0:6.9', type='build')
depends_on('libxc@4.0.3:4.99.0', when='+libxc@7.0:')
```

Reply:
Why are some libxc type=build, others are type=build/link, and others are type=link?

Reply:
The type of dpend_on has changed because the libxc library (libxcf03.so.5) is now linked from cp2k version 7.1.
I was misunderstanding that depend on should be link' if needed at link time.
I will re-correct with reference to the pointed out contents.